### PR TITLE
[ArrowStringArray] CLN: remove hasattr checks

### PR DESCRIPTION
**Pull request by simonjayhawkins**
xref https://github.com/pandas-dev/pandas/pull/41281#discussion_r625926407

### CODE REVIEW

File: pandas/compat/__init__.py [hunk 1]
```diff
@@ -129,11 +130,31 @@ def get_lzma_file(lzma):
     return lzma.LZMAFile
 
 
+# pyarrow versioning
```
Comment (jreback) [inline pandas/compat/__init__.py:133]: can you put in pandas.compat.pyarrow (and import), similar to how we do numpy

**Comment by jreback** [comment]
one more rebase if you would (just merged another arrow string PR )

**Review by jreback — APPROVED** [review]
lgtm merge on green

**Comment by simonjayhawkins** [comment]
> lgtm merge on green

only codecov/patch failing so merging

**Pull request merged**